packages/ckeditor5-dev-release-tools/lib/tasks/releasesubrepositories.js

```diff
@@ -727,10 +727,17 @@ module.exports = async function releaseSubRepositories( options ) {
 			// For this reason we have to temporarily replace the extension in the `main` field while the package is being published to npm.
 			// This change is then reverted.
 			const hasTypeScriptEntryPoint = packageJson.main && packageJson.main.endsWith( '.ts' );
+			const hasTypesProperty = !!packageJson.types;
 
 			if ( hasTypeScriptEntryPoint ) {
 				tools.updateJSONFile( packageJsonPath, jsonFile => {
-					jsonFile.main = jsonFile.main.replace( '.ts', '.js' );
+					const { main } = jsonFile;
+
+					jsonFile.main = main.replace( /\.ts$/, '.js' );
+
+					if ( !hasTypesProperty ) {
+						jsonFile.types = main.replace( /\.ts$/, '.d.ts' );
```

Comment:
All looks fine, just a loose thought. Is it a problem when we specify `"types": "index.d.ts"` when the `index.d.ts` file does not exist? Probably this is not an issue, but maybe it's worth checking.

Reply:
I added a check.

![image](https://user-images.githubusercontent.com/2270764/221505693-117e0cdc-3eef-4eb3-8828-96c79e39a009.png)

I agree it is not the best design here, but let's be safe than sorry.